handlers/media.go

```diff
@@ -0,0 +1,150 @@
+package handlers
+
+import (
+	"context"
+	"strings"
+
+	"github.com/nyaruka/courier"
+	"github.com/pkg/errors"
+)
+
+type MediaType string
+
+const (
+	MediaTypeImage       MediaType = "image"
+	MediaTypeAudio       MediaType = "audio"
+	MediaTypeVideo       MediaType = "video"
+	MediaTypeApplication MediaType = "application"
+)
+
+type MediaTypeSupport struct {
+	Types    []string
+	MaxBytes int
+}
+
+// Attachment is a resolved attachment
+type Attachment struct {
+	Type        MediaType
+	ContentType string
+	URL         string
+	Media       courier.Media
+	Thumbnail   courier.Media
+}
+
+// ResolveAttachments resolves the given attachment strings (content-type:url) into attachment objects
+func ResolveAttachments(ctx context.Context, b courier.Backend, attachments []string, support map[MediaType]MediaTypeSupport, allowURLOnly bool) ([]*Attachment, error) {
+	resolved := make([]*Attachment, 0, len(attachments))
+
+	for _, as := range attachments {
+		att, err := resolveAttachment(ctx, b, as, support, allowURLOnly)
+		if err != nil {
+			return nil, err
+		}
+		if att != nil {
+			resolved = append(resolved, att)
+		}
```

Comment:
Allowing attachments to be skipped over if they're not supported... debatable what we should do in this case? Could just error the message too..

Reply:
I guess skipping over attachments is at least consistent with how it works currently if a channel handler just does't support sending attachments.

Reply:
I thought when a media is over the limit of the max supported by a channel type we show a channel log with that error specified

Reply:
So telegram send an image well but WA fail because that image is bigger than the max for WA

Reply:
Here I think we are silently skipping the attachment

Reply:
@rowanseymour have you seen this?

Reply:
Yeah.. I'm not keen on adding more reasons to create channel logs, until we can rework channel logs in general.

I appreciate this might be confusing but it's no more confusing than the current behavior which is silently skipping if the channel type doesn't support attachments.

Reply:
I was thinking now we try to send the the channel API the file they do not support and then that is rejected and we have a log now that we see the response form the API has rejected the file type.

Now here I think we are not at all making the request and fail and we silently skip the attachment.

My understanding is that we can create the log without making the API request, but yes we can add that later when the logs are reworked.